### PR TITLE
fixes #3430 ACP settings search option hardcoded only for english language

### LINKED ISSUE #3430: ACP settings search option - hardcoded only for english language.
myBB version :
1.8.18

Issue :
search  in ACPs configuration tab is "hardcoded" to always use english language, independently from language chosed by administrator. Therefore search does not find anything when administrator use different language than english.

Steps to reproduce :
1. Goto ACP
2. Change ACP language to anything but english.
3. Click "configuration" tab at top of page or go to admin/index.php?module=config-settings
4. Insert non english word in search textbox and click search.
5. Obviously nothing will be found.

Details :
admin\modules\config\settings.php at line 1147 perform search by query on DB data which is in english.
It should rather create an array of translated data and search this instead. (or similar ideas)

Original thread: [ACP settings search option - hardcoded only for english language.](https://community.mybb.com/thread-219527.html)

### PR DESCRIPTION
**Pull request by avril-gh**
Fixes issue where ACP settings search option, has searched directly DB data (which is in english) without considering translation. Because of this, ACP options search feature didnt worked for any other language than english.

### CODE REVIEW

**Comment by avril-gh** [comment]
fixes #3430

**Comment by WildcardSearch** [comment]
@avril-gh I have a little time to test this week.

Make sure I have steps to reproduce correct:

- Load ACP using a translation
- ACP->Config->Settings [search box]
- current behavior (doesn't find setting because it is looking for English name)
- expected behavior (searches for translated language when appropriate)

**Comment by avril-gh** [comment]
If admin use in ACP another language than english, he obviously will name options in his own language, but since search box directly query on DB tables which are in english. he will not find the option he is looking for.

The solution is to not search directly on DB tables (like now) but instead, fetch them and search them localy while applying translation in the fly.

like usual, tested closely at each stage before submitting PR
Thanks for attention @WildcardSearch

**Comment by WildcardSearch** [comment]
Thank you, @avril-gh.

I was able to reproduce the error. Your PR solves the issue. Maybe @dvz or @euantorano will have time to merge it.

File: admin/modules/config/settings.php [hunk 1]
```diff
@@ -1145,18 +1145,41 @@
 		// Search for settings
 		$search = $db->escape_string_like($mybb->input['search']);
 		$query = $db->query("
-			SELECT s.*
+			SELECT s.* , g.name as gname, g.title as gtitle, g.description as gdescription
```
Comment (avril-gh) [inline admin/modules/config/settings.php]: we'll fetch all settings

**Comment by WildcardSearch** [comment]
Fixes #3430

**Pull request merged**